sdk/src/Handlers/AspNetCore/Internal/AWSXRayMiddleware.cs

```diff
@@ -273,9 +275,32 @@ private static void PopulateRequestAttributes(HttpRequest request, Dictionary<st
             }
         }
 
-        private static string GetUrl(HttpRequest request)
+        // Implementing custom logic : https://github.com/aws/aws-xray-sdk-dotnet/issues/64
+        public static string GetUrl(HttpRequest request)
         {
-            return Microsoft.AspNetCore.Http.Extensions.UriHelper.GetDisplayUrl(request);
```

Comment:
From my understanding, in stead of calling `GetDisplayUrl`, you implementing a way to construct the url by itself? Why you don't want to use the method when the request is not NULL.

Reply:
This is related to this [issue](https://github.com/aws/aws-xray-sdk-dotnet/issues/64) 

The NET Core 2.x throws null pointer exception on `GetDisplayUrl()` call, when `Host` header is not present in the incoming request. This is a bug from the downstream dependency which is fixed in NET Core 3.x (not yet GAed).

I talked to ASP NET Core community and they provided to copy paste the code : [link](https://github.com/aspnet/AspNetCore/issues/2718), see my comments in the last section. 

The request is not null is just a safety check, however the issue lies deeper inside the `request` object. Looking at the current situation, writing custom code gives fine grain control on null safety checks on individual parameters of the request instance.

Reply:
I see. Once they fix the bug, are we going to keep our own implemented logic or using the new one they implemented?

Reply:
Once the ASP.NET Core back ports the change from NET Core 3.x to 2.x, I will upgrade the ASP.NET Core dependency and remove the custom logic. However the ETA is late June.

Reply:
Cool, make sure we add this into our backlog to upgrade the new dependency once they fix it.